src/main/java/com/acme/dbo/txlog/Facade.java

```diff
@@ -1,11 +1,41 @@
 package com.acme.dbo.txlog;
 
 public class Facade {
+    public static boolean isDecorated;
+
+    private static void decorateLogPrimitive(String message) {
+        if (Facade.isDecorated) {
+            System.out.println("primitive: " + message);
+        } else {
+            System.out.println(message);
+        }
+    }
+
     public static void log(int message) {
-        System.out.println("primitive: " + message);
+        Facade.decorateLogPrimitive(String.valueOf(message));
+    }
+
+    public static void log(boolean message) {
+        Facade.decorateLogPrimitive(String.valueOf(message));
+    }
+
+    public static void log(char message) {
+        if (Facade.isDecorated) {
+            System.out.println("char: " + message);
+        } else {
+            System.out.println(message);
+        }
+    }
+
+    public static void log(String message) {
+        if (Facade.isDecorated) {
```

Comment:
Why use _isDecorated_ variable as static? For now you make your _global application state_ contain decoration flag. Is it really needed? Can you pass this flag either as parameter to printing methods or making special method?
`log(PRIMITIVE_DECORATION_PREIX, message)` or `logPrimitive(message)` with use of PRIMITIVE_DECORATION_PREIX inside?

Reply:
You've mentioned that Logger becomes stateful. What should be considered as a unavoidable state?

Offtopic question but I don't know how to ask:
How can I log the last value in task "shouldLogSequentIntegersAsSum" if I designed postponed logic, i.e. I print integers only when other types used. I don't print integers in integer log. Logic works apart from the last integer.

Reply:
> What should be considered as a unavoidable state?
Accumulators for aggregating values (from iteration 02 requirements)

Reply:
> How can I log the last value in task "shouldLogSequentIntegersAsSum"

Perfect question ) For now there is no way to do this ))
So you must have some kind of event "it's the end, flush accumulated value, I'm done."
I recommend you just add new method, like a stop() or flush()

Reply:
> > How can I log the last value in task "shouldLogSequentIntegersAsSum"
> 
> Perfect question ) For now there is no way to do this ))
> So you must have some kind of event "it's the end, flush accumulated value, I'm done."
> I recommend you just add new method, like a stop() or flush()

Where can I call this method? 
1) Test @After is too late for this because test is done at that state.
2) Inside Facade class there is no way to call it.

Reply:
iteration01Tests.java:
`
Facade.log();
Facade.flush();
Facade.log();
Facade.flush();

assertXXX();
assertYYY();
`

iteration02Tests.java:
`
Facade.log();
Facade.log();
Facade.flush();

assertXXX();
assertYYY();
`

Reply:
Done both.